### PR TITLE
feat: add LLM output for the AI ML component to use ChatOpenAI base model

### PR DESCRIPTION
**Pull request by vasconceloscezar**
The current AI ML component only outputs Text, and all other AI Models output `Text` and `Language Model`

image

With this change, we can use models from AI ML in `Agents` and other components that use the `Language Model` as input. 

image

### CODE REVIEW

**Review by ogabrielluiz — APPROVED** [review]
LGTM

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 1]
```diff
@@ -1,148 +1,128 @@
...
         ),
-        FloatInput(
-            name="repeat_penalty",
-            display_name="Repeat Penalty",
-            info="Penalty for repetitions in generated text. (Default: 1.1)",
+        StrInput(
+            name="aiml_api_base",
+            display_name="AIML API Base",
             advanced=True,
+            info="The base URL of the OpenAI API. Defaults to https://api.aimlapi.com . You can change this to use other APIs like JinaChat, LocalAI e Prem.",
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:62]: I believe the intent of this component was to only hit the AI/ML API, so would lean towards removing this parameter. If users want to use a different base, we can add another component for that or use the generic OpenAI component. (I'm glad using the `base_url` here works using the aiml base - I tried with the `OpenAIEmbeddings` class and had issues).

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 2]
```diff
@@ -1,148 +1,128 @@
...
         ),
-        FloatInput(
-            name="temperature",
-            display_name="Temperature",
-            value=0.2,
-            info="Controls the creativity of model responses.",
+        SecretStrInput(
+            name="api_key",
+            display_name="AIML API Key",
+            info="The AIML API Key to use for the OpenAI model.",
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:67]: Shouldn't be specific to `the OpenAI model`

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 3]
```diff
@@ -1,148 +1,128 @@
...
-            messages.append({"role": "system", "content": self.system_message})
+    def build_model(self) -> LanguageModel:  # type: ignore[type-var]
+        output_schema_dict: dict[str, str] = reduce(operator.ior, self.output_schema or {}, {})
+        aiml_api_key = self.api_key
+        temperature = self.temperature
+        model_name: str = self.model_name
+        max_tokens = self.max_tokens
+        model_kwargs = self.model_kwargs or {}
+        aiml_api_base = self.aiml_api_base or "https://api.aimlapi.com"
+        json_mode = bool(output_schema_dict) or self.json_mode
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:89]: Seems like `output_schema` exists only to determine whether to set `json_mode`. I do have the concern that even if fixed, the AI/ML API wouldn't support this (or, only would for specific models, which may be confusing for users).

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 4]
```diff
@@ -1,148 +1,128 @@
...
-                # Though we aren't using langchain here, the helper method is useful
-                message = self.input_value.to_lc_message()
-                if message.type == "human":
-                    messages.append({"role": "user", "content": message.content})
-                else:
-                    raise ValueError(f"Expected user message, saw: {message.type}")
-            else:
-                raise TypeError(f"Expected Message type, saw: {type(self.input_value)}")
+        if isinstance(aiml_api_key, SecretStr):
+            openai_api_key = aiml_api_key.get_secret_value()
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:93]: nit: var name should stay `aiml_api_key`

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 5]
```diff
@@ -1,148 +1,128 @@
...
-            "stop_tokens": self.stop_tokens or None,
-        }
+        model = ChatOpenAI(
+            model=model_name,
+            temperature=temperature,
+            api_key=openai_api_key,
+            base_url=aiml_api_base,
+            max_tokens=max_tokens or None,
+            seed=seed,
+            json_mode=json_mode,
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:104]: as noted - `json_mode` isn't an explicit parameter to `ChatOpenAI`

File: src/backend/base/langflow/components/models/AIMLModel.py [hunk 6]
```diff
@@ -1,148 +1,128 @@
...
-            "model": self.model_name,
-            "messages": messages,
-            "max_tokens": self.max_tokens or None,
-            "temperature": self.temperature or 0.2,
-            "top_k": self.top_k or 40,
-            "top_p": self.top_p or 0.9,
-            "repeat_penalty": self.repeat_penalty or 1.1,
-            "stop_tokens": self.stop_tokens or None,
-        }
+        model = ChatOpenAI(
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/models/AIMLModel.py:97]: Was curious about this usage - does using `OpenAI`'s implementation have any compatibility issues using other models?

**Pull request merged**